### PR TITLE
Return ID and full path of uploaded file

### PR DESCRIPTION
**Pull request by bombillazo**
## What kind of change does this PR introduce?

Bug fix for #170 

## What is the current behavior?

Currently, only part of the path (not including the bucket) of the uploaded file is returned even though the API returns both the ID and full path of the file: [source](https://supabase.github.io/storage-api/#/object/post_object__bucketName___wildcard_)

## What is the new behavior?

The new behavior is for the response of the upload client call to return the id and full path in addition to the path.

### CODE REVIEW

**Comment by bombillazo** [comment]
Hey @awalias @soedirgo @inian this is a PR for your consideration, would help us greatly to adopt the storage and make it easier to use via the JS client.

**Comment by bombillazo** [comment]
Anyone got a chance to check?

**Comment by bombillazo** [comment]
Any maintainer willing to help?

**Comment by fenos** [comment]
Hey @bombillazo i'll look at this ASAP

**Comment by bombillazo** [comment]
Hey @fenos any update?

**Comment by bombillazo** [comment]
Any update on this???

**Comment by bombillazo** [comment]
Any update?

**Comment by WildEgo** [comment]
> Hey @bombillazo i'll look at this ASAP

Hey mate have you had time to look into it? I'm currently in need of this feature

**Comment by bombillazo** [comment]
Nothing yet?

**Comment by bombillazo** [comment]
Please, is there any update on this?

**Comment by inian** [comment]
Hi @bombillazo, this PR needs [v3 of our storage](https://supabase.com/blog/storage-v3-resumable-uploads) to be rolled out to everyone since only v3 returns the id of the uploaded file. We still have a few regions to roll out for v3, but aiming to do that by the first week of October. We will merge the PR right after. As a workaround, you can contact support to upgrade your own projects to v3, if it hasn't already been upgraded, and use a fork version of storage-js till next month. 

Apologies for the lack of communication here!

**Comment by bombillazo** [comment]
Thanks for the updates!

**Review by fenos — APPROVED** [review]

**Pull request merged**
